### DISCUSSION TITLE
Confusion on the corrosion build dir

### DISCUSSION THREAD

**Discussion by Christian-Prather** [corrosion-rs/corrosion#572] [Q&A] [answered]
Hello, 
I am currently using Corrosion to build rust bins and I have been dealing with some build issues. In looking at the build logs I'm seeing a weird path for the build? When building a package with the top level `CMAKE_BINARY_DIR` of `/app/ws/build_debug/test_package` I am seeing that for some reason builds are occurring in the following path for a crate `OUT_PATH`

`/app/ws/build_debug/test_package/./cargo/build/x86_64-unknown-linux-gnu/debug/build/libffi-sys-4c3df4092463751b`

The weirdness is from the fact that the `OUT_PATH` has ` /./ ` in it seems to be raising alarm bells for me that something with the path setting is wrong? Any input would be amazing. 

Also note: the above is also failing to build for some reason?

**Comment by jschwe** [comment, marked as answer]
The `.` comes from here:  https://github.com/corrosion-rs/corrosion/blob/1c6974c2473765449e7c4649f9f96f1b751420c3/cmake/Corrosion.cmake#L589C14-L589C23
It just makes it slightly more convenient to construct the [cargo target directory here](https://github.com/corrosion-rs/corrosion/blob/1c6974c2473765449e7c4649f9f96f1b751420c3/cmake/Corrosion.cmake#L677), since the exact path depends on the generator.
I would accept a pull request that removes the `.` from the path

> `x86_64-unknown-linux-gnu/debug/build/libffi-sys-4c3df4092463751b` 

this part is just how cargo builds dependencies.

> Also note: the above is also failing to build for some reason?

I'm afraid that is not very helpful. You will need to provide more information than just "it fails".
Especially `-sys` crates often need some additional configuration in the form of environment variables, for which you would need to consult the documentation of the respective crate.

**Reply by Christian-Prather** [reply]
Thank you so much! That is very informative, and if I find some time Id be happy to put up a PR (not that I think its very important now that I know its by design) bur none the less happy to contribute. 

>  You will need to provide more information than just "it fails".

Yes sorry! That was not meant to be a "help me fix this" comment just added for context in case it related to the `/./` path. While I am having trouble with the libffi crate and was hoping to ask for some support soon, I figured I would keep the two concerns separate and provide more info once I had exhausted my own attempts. Thank you though!
